src/server/api/routers/hardware.ts

```diff
@@ -143,92 +145,111 @@ export const hardwareRouter = createTRPCRouter({
     .use(workspaceAccessMiddleware)
     .output(hardware)
     .mutation(async ({ ctx, input }) => {
+      // TODO: Not working for now
       return await ctx.db.transaction().execute(async (tx) => {
-        const { components, ...newHardware } = input;
-        const hardware = await tx
-          .insertInto("hardware")
-          .values({
-            id: generateDatabaseId("hardware"),
-            ...newHardware,
-          })
-          .returningAll()
-          .executeTakeFirstOrThrow(
-            () =>
-              new TRPCError({
-                code: "INTERNAL_SERVER_ERROR",
-                message: "Failed to create hardware",
-              }),
-          );
-        const model = await getModelById(hardware.modelId);
-        if (!model) {
-          throw new TRPCError({
-            code: "BAD_REQUEST",
-            message: "Model not found",
-          });
-        }
-
-        const modelComponents = await getModelComponents(model.id);
-
-        if (modelComponents.length > 0) {
-          const ids = components.map((c) => c.hardwareId);
-          if (_.uniq(ids).length !== ids.length) {
+        try {
```

Comment:
should not just wrap the entire thing in a try catch, should handle each error individually :)